core/imt/inc/ROOT/RSlotStack.hxx

```diff
@@ -18,10 +18,15 @@
 namespace ROOT {
 namespace Internal {
 
-/// This is an helper class to allow to pick a slot resorting to a map
-/// indexed by thread ids.
-/// WARNING: this class does not work as a regular stack. The size is
-/// fixed at construction time and no blocking is foreseen.
+/// A thread-safe stack of N indexes (0 to size - 1).
+/// RSlotStack can be used to safely assign a "processing slot" number to
+/// each thread in multi-thread applications.
+/// In release builds, pop and push operations are unchecked, potentially
+/// resulting in undefined behavior if more slot numbers than avaialble are
+/// requested.
```

Comment:
```suggestion
/// accessed.
```

Reply:
slot numbers are not accessed  you can request a new slot number from the `RSlotStack` with `GetSlot` and that pops a number out of a `std::stack<unsigned int>`.